extensions/functions_string.yaml

```diff
@@ -163,3 +165,236 @@ scalar_functions:
+          - value: "fixedchar<L2>"
+            name: "characters"
+            description: "The set of characters to remove."
+        return: "string"
+  -
+    name: lpad
+    description: Left-pad the string with the characters until the specified length of the string has been reached.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "varchar<L2>"
+            name: "characters"
+            description: "The set of characters to use for padding."
+        return: "string"
+      - args:
+          - value: "string"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "string"
+            name: "characters"
+            description: "The set of characters to use for padding."
+        return: "string"
+      - args:
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "fixedchar<L2>"
+            name: "characters"
+            description: "The set of characters to use for padding."
+        return: "string"
+  -
+    name: rpad
+    description: Right-pad the string with the characters until the specified length of the string has been reached.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "varchar<L2>"
+            name: "characters"
+            description: "The set of characters to use for padding."
+        return: "string"
+      - args:
+          - value: "string"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "string"
+            name: "characters"
+            description: "The set of characters to use for padding."
+        return: "string"
+      - args:
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: "The string to pad."
+          - value: i32
+            name: "length"
+            description: "The length of the output string."
+          - value: "fixedchar<L2>"
+            name: "characters"
+            description: "The set of characters to use for padding."
+        return: "string"
+  -
+    name: center
+    description: Pad the string with characters from each side until the specified length of the string has been reached.
```

Comment:
Would this need an option to determine which side to add more items to if the number of character to add was an odd number? Or should the consumer just decide that?

Reply:
I'm not actually sure either.  From what i've seen, the default is that for uneven padding, the lesser amount of padding goes on the left.  I can at least specify that in the description.

Reply:
So the reason I brought it up is that when I was writing R bindings for Arrow, this came up and I think we ended up requesting it being added as an option in Acero as the R library handled  it differently to Acero. Not sure how common this is though - definitely worth checking as it's a pain to work around.